pkg/analysis/analysis.go

```diff
@@ -175,11 +176,10 @@ func (a *Analysis) RunAnalysis() []error {
 					<-semaphore
 				}(analyzer, filter)
 			} else {
-				errorList = append(errorList, fmt.Errorf(fmt.Sprintf("\"%s\" filter does not exist. Please run k8sgpt filters list.", filter)))
+				a.Errors = append(a.Errors, fmt.Sprintf(fmt.Sprintf("\"%s\" filter does not exist. Please run k8sgpt filters list.", filter)))
```

Comment:
It occurs to me this needs a mutex around it

Reply:
I think, it is not necessary to use mutex.Lock() and mutex.Unlock() around the else statement. Since there is no concurrent access or modification of shared variables in the else block, you don't need to use a mutex lock there. What do you think ?